Client/core/CSettings.cpp

```diff
@@ -1004,12 +1004,26 @@ void CSettings::CreateGUI()
     m_pAdvancedMiscLabel->AutoSize();
     vecTemp.fY += fHeaderHeight;
 
-    fIndentX = pManager->CGUI_GetMaxTextExtent("default-normal", _("Fast CJ clothes loading:"), _("Browser speed:"), _("Single connection:"), _("Packet tag:"),
+    fIndentX = pManager->CGUI_GetMaxTextExtent("default-normal", _("Radar map image:"), _("Fast CJ clothes loading:"), _("Browser speed:"), _("Single connection:"), _("Packet tag:"),
                                                _("Progress animation:"), _("Fullscreen mode:"), _("Process priority:"), _("Debug setting:"),
                                                _("Streaming memory:"), _("Update build type:"), _("Install important updates:")) +
                5.0f;
 
     vecTemp.fX += 10.0f;
+
+    // Radar map image
+    m_pRadarMapImageLabel = reinterpret_cast<CGUILabel*>(pManager->CreateLabel(pTabAdvanced, _("Radar map image:")));
+    m_pRadarMapImageLabel->SetPosition(CVector2D(vecTemp.fX, vecTemp.fY));
+    m_pRadarMapImageLabel->AutoSize();
+
+    m_pRadarMapImageCombo = reinterpret_cast<CGUIComboBox*>(pManager->CreateComboBox(pTabAdvanced, ""));
+    m_pRadarMapImageCombo->SetPosition(CVector2D(vecTemp.fX + fIndentX, vecTemp.fY - 1.0f));
+    m_pRadarMapImageCombo->SetSize(CVector2D(fComboWidth, 95.0f));
+    m_pRadarMapImageCombo->AddItem(_("1024 x 1024 (Default)"))->SetData((void*)0);
+    m_pRadarMapImageCombo->AddItem(_("2048 x 2048"))->SetData((void*)1);
```

Comment:
what about a constexpr array with resolutions?

also `(void*)0` and `(void*)1` looks very dangerous. What exactly the `SetData` function expects?

Reply:
Defining resolutions as an array using `constexpr ` is a great idea for both readability and maintainability. Using `(void*)0` and `(void*)1` can be dangerous because these types of pointer conversions are error-prone and difficult to understand. Using a `constexpr ` array and an enum to express the intent in the code would be much safer and clearer.

Reply:
![image](https://github.com/user-attachments/assets/23c1ff04-e512-468c-a02a-90a0a415c26c)

i copied the others, hmmm let me check.